shared/pingpong/pingpong.go

```diff
@@ -150,6 +151,14 @@ func RunPingPong(ctx context.Context, ac libgoal.Client, accounts map[string]uin
 
 				refreshTime = refreshTime.Add(cfg.RefreshTime)
 			}
+
+			localTimeDelta := time.Now().Sub(startTime)
+			currentTps := float64(totalSent) / localTimeDelta.Seconds()
+			if currentTps > float64(cfg.TxnPerSec) {
+				sleepSec := float64(totalSent)/float64(cfg.TxnPerSec) - localTimeDelta.Seconds()
+				sleepTime := time.Duration(int64(math.Round(sleepSec*1000))) * time.Millisecond
+				time.Sleep(sleepTime)
+			}
```

Comment:
That code would only work to slow down a single txn time to match the desired txn/sec time.
Unfortunately, when we try to push it to the end, we get to uneven txn times, where this wouldn't work so well.
I suggest that you'll try to implement it using a moving window of size cfg.TxnPerSec*2.